src/commands/ping/ping.ts

```diff
@@ -1,21 +1,28 @@
 import { ApplyOptions } from '@sapphire/decorators';
-import { Command, CommandOptions, container } from '@sapphire/framework';
-import type { Message } from 'discord.js';
+import { Command, container } from '@sapphire/framework';
+import { Message } from 'discord.js';
+import { APIMessage } from 'discord-api-types/v9';
+import { isMessageInstance } from '@sapphire/discord.js-utilities';
 
-@ApplyOptions<CommandOptions>({
-  aliases: ['pong'],
+@ApplyOptions<Command.Options>({
+  aliases: ['ping'],
   description: 'ping pong',
   detailedDescription: `**Examples:**
-\`${container.botPrefix}ping\`
-\`${container.botPrefix}pong\``
+    \`${container.botPrefix}ping\`
+    \`${container.botPrefix}pong\``,
+  chatInputCommand: {
+    register: true
+  }
 })
 export class PingCommand extends Command {
-  async messageRun(message: Message): Promise<Message> {
-    const { client } = container;
-    const msg = await message.channel.send('Ping?');
-    const content = `Pong from JavaScript! Bot Latency ${Math.round(client.ws.ping)}ms. API Latency ${
-      msg.createdTimestamp - message.createdTimestamp
-    }ms.`;
-    return msg.edit(content);
+  public async chatInputRun(interaction: Command.ChatInputInteraction): Promise<APIMessage | Message<boolean>> {
```

Comment:
this removes the message command (with .ping). We should probs keep this at least until every command is slash.

Reply:
It seems like using slash commands makes regular commands not work.

Reply:
This is now done. I tried to abstract as much as possible - let me know what you think.